packages/analytics-js-plugins/src/beaconQueue/index.ts

```diff
@@ -75,12 +81,20 @@ const BeaconQueue = (): ExtensionPlugin => ({
 
         if (data) {
           try {
-            const isEnqueuedInBeacon = navigator.sendBeacon(url, data);
-            if (!isEnqueuedInBeacon) {
-              logger?.error(BEACON_QUEUE_SEND_ERROR(BEACON_QUEUE_PLUGIN));
-            }
+            if (!navigator.sendBeacon(url, data)) {
+              if (isPageAccessible) {
+                logger?.error(
+                  `${BEACON_QUEUE_SEND_ERROR(BEACON_QUEUE_PLUGIN)} The event(s) will be dropped.`,
+                );
 
-            done(null, isEnqueuedInBeacon);
+                // Remove the item from queue
+                done(null);
+              } else {
+                logger?.warn(
+                  `${BEACON_QUEUE_SEND_ERROR(BEACON_QUEUE_PLUGIN)} The event(s) will be retried as the current page is being unloaded.`,
+                );
+              }
+            }
```

Comment:
This if block only covers the case where items can not be added to the beacon queue. But in case of successful queueing, we need to call the done() callback.

Reply:
Addressed. Thanks.